library/proc_macro/src/bridge/buffer.rs

```diff
@@ -5,38 +5,40 @@ use std::mem;
 use std::ops::{Deref, DerefMut};
 use std::slice;
 
+// `T` is always `u8` in practice. Attempts to remove `T` caused mild
+// performance regressions, surprisingly enough.
```

Comment:
Actually, every `fn` in this module might need `#[inline]` (all 17 of them), so can you just edit the comment to say that without generics, `#[inline]` would be needed everywhere to avoid performance regressions and r=me?